src/GraphUtils.ts

```diff
@@ -401,42 +397,21 @@ export function findTurnPts(pts: Point[], mode: string) {
             }
         }
         if (!isDefined(position)) continue;
-        if (statPts[i][1] < pts[position-5][1] && statPts[i][1] < pts[position+5][1]) {
+        if (statPts[i][1] < pts[mod(position-5, pts.length)][1] && statPts[i][1] < pts[mod(position+5, pts.length)][1]) {
+            // if the point we have found is within 5 units of the previous point, only include one of them
+            // if (pts.findIndex((v: Point) => _isEqual(statPts[i], v)) === 0) {
             pot_max.push(statPts[i]);
-        } else if (statPts[i][1] > pts[position-5][1] && statPts[i][1] > pts[position+5][1]) {
+        } else if (statPts[i][1] > pts[mod(position-5, pts.length)][1] && statPts[i][1] > pts[mod(position+5, pts.length)][1]) {
```

Comment:
```suggestion
        } else if (statPts[i][1] < pts[mod(position-5, pts.length)][1] && statPts[i][1] < pts[mod(position+5, pts.length)][1]) {
```
Similarly this should be a minima